File: JavaScript/src/agingStrategies/Normal.js
// Decreases an item's quality to 0 over time
export default class Normal {
  static tick(item) {
    item.daysRemaining -= 1;

    item.quality -= 1;
    if (item.daysRemaining <= 0) {
      item.quality -= 1;
    }

    item.quality = Math.max(item.quality, this.#minimumQuality());
  }

  static #minimumQuality() {
    return 0;
  }
}

File: Ruby/lib/aging_strategies/normal.rb
# frozen_string_literal: true

module AgingStrategies
  # Decreases an item's quality to 0 over time
  class Normal
    def self.tick(item)
      item.days_remaining -= 1

      item.quality -= 1
      item.quality -= 1 if item.days_remaining <= 0
      item.quality = [item.quality, minimum_quality].max
    end

    private_class_method def self.minimum_quality
      0
    end
  end
end
